js/iframe-coordinator/libs/client.ts

```diff
@@ -53,56 +96,13 @@ export default {
     message: string,
     { title = null, custom = null }: undefined | ToastOptions = {}
   ): void {
-    sendMessage("toastRequest", {
+    this._sendMessage("toastRequest", {
       title,
       message,
       custom
     });
   }
-};
-
-function start() {
-  if (!worker) {
-    worker = Elm.Client.init();
-
-    window.addEventListener("message", (event: MessageEvent) => {
-      worker.ports.fromHost.send({
-        origin: event.origin,
-        data: event.data
-      });
-    });
-
-    worker.ports.toHost.subscribe((message: any) => {
-      window.parent.postMessage(message, "*");
-    });
-
-    worker.ports.toClient.subscribe((message: any) => {
-      if (message.msgType == "publish") {
-        messageHandlers.forEach(handler => {
-          handler(message.msg);
-        });
-      }
-    });
-
-    window.addEventListener("click", event => {
-      let target = event.target as HTMLElement;
-      if (target.tagName.toLowerCase() === "a" && event.button == 0) {
-        event.preventDefault();
-        let a = event.target as HTMLAnchorElement;
-        sendMessage("navRequest", a.href);
-      }
-    });
-  }
-}
-
-function sendMessage(type: string, data: any) {
-  worker.ports.fromClient.send({
-    msgType: type,
-    msg: data
-  });
 }
 
-interface ToastOptions {
-  title?: string;
-  custom?: { [x: string]: any };
-}
+const client = new Client();
```

Comment:
Maybe we should make this `Client.ts` and export the class itself.  This is a pretty common approach and then we have the ability to instantiate multiple instances for testing, etc.